Bottom: 9e4952282ae74f8eb04d1b756c5ef9607a608d1e
Top:    a755edab2f9cd3058299b176ab8adf60ae2c0ae2
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-03 14:49:07 +0000

Fix to ParseGroupString()


---

diff --git a/src/misc.c b/src/misc.c
index 10796fc..63d7ca4 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -1187,7 +1187,14 @@ ParseGroupString (char *group_string, LayerGroupType *LayerGroup, int *LayerN)
           /* ignore white spaces and check for separator */
           while (*s && isspace ((int) *s))
             s++;
-          if (!*s || *s == ':')
+
+          /* Backup so the outer loop terminates on the NULL termination */
+          if (*s == '\0') {
+            s--;
+            break;
+          }
+
+          if (*s == ':')
             break;
         }
       LayerGroup->Number[group] = member;
